core/server/storage/local-file-store.js

```diff
@@ -51,14 +53,52 @@ LocalFileStore.prototype.exists = function (filename) {
 };
 
 // middleware for serving the files
-LocalFileStore.prototype.serve = function () {
-    // For some reason send divides the max age number by 1000
-    // Fallthrough: false ensures that if an image isn't found, it automatically 404s
-    return serveStatic(config.paths.imagesPath, {maxAge: utils.ONE_YEAR_MS, fallthrough: false});
+LocalFileStore.prototype.serve = function (options) {
+    var self = this;
+    options = options || {};
+
+    // CASE: serve themes
+    // serveStatic can't be used to serve themes, because
+    // download files depending on the route (see `send` npm module)
+    if (options.isTheme) {
```

Comment:
There was no other solution i could think of then using a `isTheme` option.
When serving a zip file, the handling is different to images. Images are served via express and this inner module `send` is download files based on the route.

So when the API serves a zip file, the storage `serve` function is responsible for checking if the zip exists and then pipe it to the client back. Because the `casper` theme has no zip file i decided to create the zip files on runtime when asking for a download. This behaviour is changeable by adding a migration to zip casper.